### ISSUE TITLE
How to load earlier messages?

### ISSUE THREAD

**Issue by Jeyhey** [MessageKit/MessageKit#184]
What is MessageKit's paradigm for loading earlier messages? I.e. how should I implement it? In JSQMessagesViewController there was a show-earlier-message-button that I could hide and show. I did not find the equivalent in MessageKit. Should I use the scroll position (reload when the first message appears)? If yes, is there a callback method for that?

**Comment by SD10** [comment]
@Jeyhey Thanks for raising this issue. We've been talking about message top loading and preserving the content offset in our Slack channel recently. Currently, we have nothing like the show earlier messages header from JSQMVC. This is on the to do list.

**Comment by Jeyhey** [comment]
Thank you for the quick response. So for the moment reloading could be implemented via pull-to-refresh or by checking visible cells when scrolling, I think.

**Comment by cwalo** [comment]
@Jeyhey I'm not sure if you caught the conversation in Slack (or elsewhere), but the process for maintaining scroll position generally goes like this:

```swift
let oldOffset = collectionView.contentOffset.y
let oldHeight = collectionView.contentSize.height
let reverseOffset = oldHeight - oldOffset

// ...decide whether to maintain the offset scrollToBottom

collectionView.reloadData()

if maintainOffset {
    collectionView.layoutIfNeeded()
    collectionView.contentOffset = CGPoint(x: 0.0, y: collectionView.contentSize.height - reverseOffset)
}
```

**Comment by nathantannar4** [comment]
This is now being looked at in PR #284

**Comment by SD10** [comment]
This will be available in `v0.12.0` through #284
